### DISCUSSION TITLE
Send image to telegram group

### DISCUSSION THREAD

**Discussion by baumrasen** [photobooth-app/photobooth-app#275] [Ideas]
Currently I use in the printing configuration a batchscript, that sends the image to a telegram group. This is working fine so far. 

The popup message "Image printed successfull" (or something like that) after successfull send the image is not so nice for that. May be it is possible to change the text for successfull "printing" so it could be used more flexible?

Is it possible, to print an image or an collage automatically? In this way all images/collages could be send to the telegram group.

**Comment by mgineer85** [comment]
Hm, maybe Printing could be replaced by something more general as "Share".
Automatic printing is not an option currently but could be added in the future.
How about using something like inotify on the media folder and invoke the script that way?
https://unix.stackexchange.com/questions/323901/how-to-use-inotifywait-to-watch-a-directory-for-creation-of-files-of-a-specific

**Comment by baumrasen** [comment]
Something like "share" sounds good for me. Maybe there could be some "templates" for sharing to x, y or z... in the future.

I like the Idea, that there don't has to be an extra service to watch a folder - but I will give inotify a try.

**Comment by mgineer85** [comment]
Would you mind to share your script for others as well?

**Reply by baumrasen** [reply]
Yes of course:
[send_photo_to_tg.gz](https://github.com/photobooth-app/photobooth-app/files/15472342/send_photo_to_tg.gz)

```
#!/bin/bash
##############################################################
# use arguments:
# first: filename of the image file
# second: caption for the image in telegram
##############################################################

##############################################################
# EDIT BOT INFO BEFORE USE!
##############################################################
BOT_TOKEN="PUT THE BOT TOKEN IN HERE"
CHAT_ID="PUT THE CHAT ID IN HERE"
file_path=""

##############################################################
# Function to send a photo to Telegram
##############################################################
send_photo() {
 local file_path="$1"
 local caption="$2"
 curl -s -X POST "https://api.telegram.org/bot$BOT_TOKEN/sendPhoto" \
 -F "chat_id=$CHAT_ID" \
 -F "photo=@$file_path" \
 -F "caption=$caption"
}

##############################################################
# send the photo
##############################################################
send_photo "$1" "$2" > /dev/null 
```

I used it like this:
`/usr/local/bin/send_photo_to_tg {filename} "Here is the photo!"`

**Reply by mgineer85** [reply]
cool, thanks. this seems quite easy. Maybe we can integrate it with the revised share-functions...
